src/main/scala/mesosphere/marathon/core/task/tracker/TaskState.scala

```diff
@@ -0,0 +1,86 @@
+package mesosphere.marathon.core.task.tracker
+
+import mesosphere.marathon.Protos.MarathonTask
+import mesosphere.marathon.core.task.tracker.impl.TaskStateSerializer
+import mesosphere.marathon.state.Timestamp
+import mesosphere.marathon.tasks.TaskIdUtil
+import org.apache.mesos.{ Protos => MesosProtos }
+
+/**
+  * The state for launching a task. This might be a launched task or a reservation for launching a task or both.
+  */
+case class TaskState(
+    taskId: TaskState.TaskId,
+    agentInfo: TaskState.AgentInfo,
+    reservationWithVolume: Option[TaskState.ReservationWithVolume.type],
+    launchCounter: Long,
+    launchedTask: Option[TaskState.LaunchedTask]) {
+
+  lazy val appId = TaskIdUtil.appId(taskId.id)
```

Comment:
could also be part of TaskId -> taskId.appId
I know there are a couple of places that need this.